frontend/public/components/utils/horizontal-nav.tsx

```diff
@@ -146,7 +146,7 @@ export const NavBar = withRouter<NavBarProps>(({ pages, basePath }) => {
           'co-m-horizontal-nav-item--active': matchUrl && matchUrl.isExact,
         });
         return (
-          <li className={klass} key={name}>
+          <li className={klass} key={name} data-test-id={name}>
```

Comment:
I think you want your test ID set on the `Link` and not the `li`. The `Link` is the interactive element.

Furthermore, this test ID has potential to clash as I do not see a parent context test ID.
Either prefix it, or add a parent ID to the horizontal nav.
eg `[data-test-id="horizontal-nav-${name}"]` or `[data-test-id=horizontal-nav] [data-test-id="${name}"]`